lifecycle-operator/webhooks/pod_mutator/handlers/workload.go

```diff
@@ -0,0 +1,120 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	klcv1alpha3 "github.com/keptn/lifecycle-toolkit/lifecycle-operator/apis/lifecycle/v1alpha3"
+	apicommon "github.com/keptn/lifecycle-toolkit/lifecycle-operator/apis/lifecycle/v1alpha3/common"
+	controllercommon "github.com/keptn/lifecycle-toolkit/lifecycle-operator/controllers/common"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type WorkloadHandler struct {
+	Client      client.Client
+	Log         logr.Logger
+	Tracer      trace.Tracer
+	EventSender controllercommon.IEvent
+}
+
+func (a *WorkloadHandler) Handle(ctx context.Context, pod *corev1.Pod, namespace string) error {
+
+	ctx, span := a.Tracer.Start(ctx, "create_workload", trace.WithSpanKind(trace.SpanKindProducer))
+	defer span.End()
+
+	newWorkload := generateWorkload(ctx, pod, namespace)
+	newWorkload.SetSpanAttributes(span)
+
+	a.Log.Info("Searching for workload")
+
+	workload := &klcv1alpha3.KeptnWorkload{}
+	err := a.Client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: newWorkload.Name}, workload)
+	if errors.IsNotFound(err) {
+		return a.createWorkload(ctx, newWorkload, span)
+	}
+
+	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
+		return fmt.Errorf("could not fetch Workload"+": %+v", err)
+	}
+
+	return a.updateWorkload(ctx, workload, newWorkload, span)
+}
+
+func (a *WorkloadHandler) updateWorkload(ctx context.Context, workload *klcv1alpha3.KeptnWorkload, newWorkload *klcv1alpha3.KeptnWorkload, span trace.Span) error {
```

Comment:
let's try to add some unit tests for the functions in this file (other than `Handle()`)